### DISCUSSION TITLE
Exporting comments

### DISCUSSION THREAD

**Discussion by dashohoxha** [posativ/isso#790] [Q&A]
Suppose that I want to move to another commenting system, how do I export the existing comments?
Maybe this seems trivial because I own the database and I can query it, but I don't have any experience with sqlight. Instead, it should not be very difficult to create a script that exports all the comments in XML or JSON format.

**Comment by posativ** [comment]
Duplicate of #183.

**Comment by dashohoxha** [comment]
No it is not.
I mean a dump of the database in XML or JSON or CSV format (if possible).

**Comment by jGleitz** [comment]
To export the comments (with thread information) into a file called `export.csv` in bash, use:

``` bash
echo -e '"page: URI","page: title","ID","mode","created on","modified on","author: name","author: email","author: website","author: IP","likes","dislikes","voters","text"\n'"$(sqlite3 /path/to/your/isso.db -csv 'SELECT threads.uri, threads.title, comments.id, comments.mode, datetime(comments.created, "unixepoch", "localtime"), datetime(comments.modified, "unixepoch", "localtime"), comments.author, comments.email, comments.website, comments.remote_addr, comments.likes, comments.dislikes, comments.voters,comments.text FROM comments INNER JOIN threads ON comments.tid=threads.id')" > export.csv
```

Don’t forget to put the correct path to your database in!

If you want the UNIX timestamps instead of a datetime string, omit the `datetime` function:

``` bash
echo -e '"page: URI","page: title","ID","mode","created on","modified on","author: name","author: email","author: website","author: IP","likes","dislikes","voters","text"\n'"$(sqlite3 /path/to/your/isso.db -csv 'SELECT threads.uri, threads.title, comments.id, comments.mode, comments.created, comments.modified, comments.author, comments.email, comments.website, comments.remote_addr, comments.likes, comments.dislikes, comments.voters,comments.text FROM comments INNER JOIN threads ON comments.tid=threads.id')" > export.csv
```

**Comment by angristan** [comment]
FWIW, I made this: https://github.com/angristan/isso-to-disqus. It exports the comments to the Disqus XML format, but can be easily adapted for something else.
